File: packages/nlu/kotlin/nlukit/src/main/kotlin/com/bridgething/nlukit/NluModelRunning.kt
package com.bridgething.nlukit

data class NluModelOutputs(
    val intentLogits: List<Float>,
    val oodLogit: Float,
    val bioLogits: List<Float>,
    val closedLogits: List<List<Float>>,
)

interface NluModelRunning {
    fun predict(inputIds: List<Int>, attentionMask: List<Int>): NluModelOutputs
}

interface NluModelPrewarming {
    fun prewarm()
}

File: packages/nlu/swift/Sources/BridgethingNluKit/NluModelRunning.swift
import Foundation

public struct NluModelOutputs: Sendable {
    public let intentLogits: [Float]
    public let oodLogit: Float
    public let bioLogits: [Float]
    public let closedLogits: [[Float]]

    public init(intentLogits: [Float], oodLogit: Float, bioLogits: [Float], closedLogits: [[Float]]) {
        self.intentLogits = intentLogits
        self.oodLogit = oodLogit
        self.bioLogits = bioLogits
        self.closedLogits = closedLogits
    }
}

public protocol NluModelRunning: Sendable {
    func predict(inputIds: [Int32], attentionMask: [Int32]) throws -> NluModelOutputs
}
